internal/server/config.go

```diff
@@ -725,3 +593,146 @@ func loadGrant(db *gorm.DB, input Grant) (*models.Grant, error) {
 
 	return grant, nil
 }
+
+func loadIdentities(db *gorm.DB, identities []Identity) error {
+	keep := make([]uid.ID, 0, len(identities)+1)
+
+	for _, i := range identities {
+		if i.Name == models.InternalInfraConnectorIdentityName {
+			return fmt.Errorf("cannot use reserved identity name %q", i.Name)
+		}
+
+		identity, err := loadIdentity(db, i)
+		if err != nil {
+			return err
+		}
+
+		keep = append(keep, identity.ID)
+	}
+
+	identity, err := loadIdentity(db, Identity{Name: models.InternalInfraConnectorIdentityName})
+	if err != nil {
+		return err
+	}
+
+	keep = append(keep, identity.ID)
+
+	if err := data.DeleteIdentities(db, data.NotIDs(keep), data.CreatedBy(models.CreatedBySystem)); err != nil {
+		return err
+	}
```

Comment:
I'm still weary of deleting things just because they aren't in the config anymore. I know that's expected for some tools, but I think it's a bit strange for API-drive systems.

The config is a way of bootstrapping entities, it's not the primary mechanism for managing entities.

Reply:
Not sure I agree with this. I would expect any resource created by configs to be removed if they're left out of a config update. I can see where you're coming from but I don't know if keeping the resource is expected

Reply:
as long as `data.CreatedBy(models.CreatedBySystem)` always points to identities created by the config and not to ones created by the API, we should be fine here. 

I sort of side with daniel on this though: If you're doing IAC it's going to be weird to have a UI/API where you can't change anything (because it's always overwritten by config). I'd really like these sources to play nice together.

Reply:
I think there are a few difference between IAC tools like `helm` and `terraform` and what we have here.

With `helm` and `terraform`:
1. the config is exclusively the API operations you want to perform
2. the config is read and applied explicitly for that reason
3. the config is read by a client and sent to an API server as API requests

With our server config:
1. it's a mix of operational settings and API operations
2. the server could be restarted for many other reasons, not necessarily because someone wants to make changes to the entities
3. the config is read by the server when a process starts or restarts

I think it would be a lot safer to print a warning in the logs, saying "entity X was removed from the config, if you no longer want it then delete it using the CLI or API", rather than deleting things.

I think it's too easy to accidentally start the process with the wrong config, and cause unintended problems. That's going to be even more of a risk when someone runs an external postgresql DB and multiple replicas of the server.

In the future if we have an `infra apply` command that reads a config to apply it as API calls, that would be a safer way to manage these entities as code.

Reply:
It's also quite possible I'm overestimating the risk, and that as long as we document it properly people will be careful.